docs/sources/mimir/manage/mimir-runbooks/_index.md

```diff
@@ -1438,6 +1442,44 @@ How to **investigate**:
 - Check wait latency of requests with strong-consistency on `Mimir / Queries` dashboard.
 - Check if ingester needs to process too many records, and whether ingesters need to be scaled up (vertically or horizontally).
 
+### Ingester is overloaded when consuming from Kafka
+
+This runbook covers the case an ingester is overloaded when ingesting metrics data (consuming) from Kafka.
+
+For example, if the amount of active series written to a partition exceeds the ingester capacity, the write-path will keep writing to the partition, but then the ingesters owning that partition will fail ingesting the data. Possible symptoms of this situation:
+
+- The ingester is lagging behind replaying metrics data from Kafka, and [`MimirStartingIngesterKafkaReceiveDelayIncreasing`](#MimirStartingIngesterKafkaReceiveDelayIncreasing) or [`MimirRunningIngesterReceiveDelayTooHigh`](#MimirRunningIngesterReceiveDelayTooHigh) alerts get fired.
+- The ingester logs [`err-mimir-ingester-max-series`](#err-mimir-ingester-max-series) when ingesting metrics data from Kafka.
+- The ingester is OOMKilled.
+
+How it **works**:
+
+- An ingester owns 1 and only 1 partition. A partition can be owned by multiple ingesters, but each ingester always own a single partition.
+- Metrics data is written to a partition by distributors, and the amount of written data is driven by the incoming traffic in the write-path. Distributors don't know whether the per-partition load is "too much" for the ingesters that will consume from that partition.
+- Ingesters are expected to autoscale. When the number of active series in ingesters grow above the scaling threshold, more ingesters will be added to the cluster. When ingesters are scaled out, new partitions are added and incoming metrics data re-balanced between partitions. However, the old data (already written to partitions) will not be moved, and the load will be re-balanced only for metrics data ingested after the scaling.
+
+How to **fix**:
+
+- **Vertical scale ingesters** (no data loss)
+  - Add more CPU/memory/disk to ingesters, depending on the saturated resources.
+  - Increase the ingester max series instance limit (see [`MimirIngesterReachingSeriesLimit`](#MimirIngesterReachingSeriesLimit) runbook).
+- **Skip replaying overloading backlog from partition** (data loss)
+  1. Ensure ingesters have been scaled out, and the new partitions are ACTIVE in the partitions ring. If autoscaler didn't scaled out ingesters yet, manually add more ingester replicas (e.g. increasing HPA min replicas or manually setting the desired number of ingester replicas).
```

Comment:
> manually setting the desired number of ingester replicas

What exactly is your suggestion here? If you set it on ingester statefulset, then we need to stop rollout-operator, or it will detect that there are too many ingesters compared to `ReplicaTemplates` object, and will start scaledown procedure.

If you set desired replica on `ReplicaTemplates` object -- I've tried this once, and HPA was very confused and scaled down my `ReplicaTemplates` object to minimum instances. (Don't know why.)

I think realistically the safest option is to set min replicas in `ScaledObject` resource.

Reply:
>manually setting the desired number of ingester replicas

I just meant for the case ingester autoscaling is disabled. I will clarify it.

Reply:
Done in https://github.com/grafana/mimir/pull/8078/commits/4e4a621fdf8a131a32a7bfe5b9ee92bd2a06853a.